alicloud/resource_alicloud_network_acl_attachment.go

```diff
@@ -0,0 +1,156 @@
+	request.Resource = &resources
+	raw, err := client.WithVpcClient(func(vpcClient *vpc.Client) (interface{}, error) {
+		return vpcClient.AssociateNetworkAcl(request)
+	})
+	if err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, "alicloud_network_acl_attachment", request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+	addDebug(request.GetActionName, raw)
+	d.SetId(request.NetworkAclId)
+
+	if err := networkAclService.WaitForNetworkAclAttachment(request.NetworkAclId, vpcResource, Available, DefaultTimeout); err != nil {
+		return WrapError(err)
+	}
+	return resourceAliyunNetworkAclAttachmentRead(d, meta)
+}
+
+func resourceAliyunNetworkAclAttachmentRead(d *schema.ResourceData, meta interface{}) error {
+
+	client := meta.(*connectivity.AliyunClient)
+	networkAclService := NetworkAclService{client}
+
+	object, err := networkAclService.DescribeNetworkAcl(d.Id())
+	vpcResource := []vpc.Resource{}
+	for _, e := range d.Get("resources").([]interface{}) {
+		res := e.(map[string]interface{})
+		vpcSource := vpc.Resource{
+			ResourceId:   res["resource_id"].(string),
+			ResourceType: res["resource_type"].(string),
+		}
+		vpcResource = append(vpcResource, vpcSource)
+	}
+	err = networkAclService.DescribeNetworkAclAttachment(object.NetworkAclId, vpcResource)
+	if err != nil {
+		if NotFoundError(err) {
+			d.SetId("")
+			return nil
+		}
+		return WrapError(err)
+	}
+	d.Set("network_acl_id", object.NetworkAclId)
+	d.Set("resources", vpcResource)
+	return nil
+}
+
+func resourceAliyunNetworkAclAttachmentDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	networkAclService := NetworkAclService{client}
+
+	resources := []vpc.UnassociateNetworkAclResource{}
+	vpcResource := []vpc.Resource{}
+	for _, e := range d.Get("resources").([]interface{}) {
+		res := e.(map[string]interface{})
+		source := vpc.UnassociateNetworkAclResource{
+			ResourceId:   res["resource_id"].(string),
+			ResourceType: res["resource_type"].(string),
+		}
+		resources = append(resources, source)
+		vpcSource := vpc.Resource{
+			ResourceId:   res["resource_id"].(string),
+			ResourceType: res["resource_type"].(string),
+		}
+		vpcResource = append(vpcResource, vpcSource)
+	}
+	request := vpc.CreateUnassociateNetworkAclRequest()
+	request.NetworkAclId = d.Id()
+	request.Resource = &resources
+	err := resource.Retry(5*time.Minute, func() *resource.RetryError {
+		raw, err := client.WithVpcClient(func(vpcClient *vpc.Client) (interface{}, error) {
+			return vpcClient.UnassociateNetworkAcl(request)
+		})
+		//Waiting for unassociate the network acl
+		if err != nil {
+			if IsExceptedError(err, TaskConflict) {
+				return resource.RetryableError(err)
+			}
+		}
+		addDebug(request.GetActionName, raw)
+		return nil
+	})
+	err = networkAclService.WaitForNetworkAclAttachment(d.Id(), vpcResource, Deleted, DefaultTimeout)
+	if err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, d.Id(), request.GetActionName(), AlibabaCloudSdkGoERROR)
```

Comment:
If the attachment has been deleted, will the method `UnassociateNetworkAcl` return an error, like "Instance not found"? If so, please check it in the lien 153